Build with cgo, required to fix build with -buildmode=pie.

See: https://gitlab.alpinelinux.org/alpine/aports/-/issues/15809

diff -upr json_exporter-0.6.0.orig/Makefile.common json_exporter-0.6.0/Makefile.common
--- json_exporter-0.6.0.orig/Makefile.common	2024-03-11 11:30:52.267097440 +0100
+++ json_exporter-0.6.0/Makefile.common	2024-03-11 11:31:01.413790555 +0100
@@ -197,7 +197,7 @@ common-unused:
 .PHONY: common-build
 common-build: promu
 	@echo ">> building binaries"
-	$(PROMU) build --prefix $(PREFIX) $(PROMU_BINARIES)
+	$(PROMU) build --cgo --prefix $(PREFIX) $(PROMU_BINARIES)
 
 .PHONY: common-tarball
 common-tarball: promu
